src/components/ShareMenu/ShareMenu.js

```diff
@@ -44,38 +100,62 @@ const defaultProps = {
       className: 'ta-mail-icon',
     },
   ],
-  extraSocialShareConfig: [],
+  extraSocialShareConfig: [
+    {
+      title: 'Karte als Bild speichern.',
+      icon: <TiImage focusable={false} />,
+      className: 'ta-image-icon',
+      saveFormat: 'image/jpeg',
+    },
+  ],
   t: t => t,
 };
 
-const ShareMenu = ({
-  className,
-  classNameIcon,
-  socialShareConfig,
-  extraSocialShareConfig,
-  t,
-}) => {
-  const config = [...socialShareConfig, ...extraSocialShareConfig];
+class ShareMenu extends PureComponent {
+  render() {
+    const {
+      map,
+      className,
+      classNameIcon,
+      socialShareConfig,
+      extraSocialShareConfig,
+      t,
+    } = this.props;
 
-  for (let i = 0; i < config.length; i += 1) {
-    config[i].url = config[i].url.replace('{url}', window.location.href);
-  }
+    const config = [...socialShareConfig, ...extraSocialShareConfig];
 
-  return (
-    <div className={className}>
-      {config.map(conf => (
-        <div
-          className={`${classNameIcon} ${conf.className || ''}`}
-          key={conf.title}
-        >
-          <BlankLink href={conf.url} title={t(conf.title)}>
-            {conf.icon}
-          </BlankLink>
-        </div>
-      ))}
-    </div>
-  );
-};
+    for (let i = 0; i < config.length; i += 1) {
+      if (config[i].url) {
+        config[i].url = config[i].url.replace('{url}', window.location.href);
+      }
+    }
+
+    return (
+      <div className={className}>
+        {config.map(conf => (
+          <div
+            className={`${classNameIcon} ${conf.className || ''}`}
+            key={conf.title}
+          >
+            {conf.url ? (
+              <BlankLink href={conf.url} title={t(conf.title)}>
+                {conf.icon}
+              </BlankLink>
+            ) : (
+              <CanvasSaveButton
```

Comment:
Something bothering me here :) if we add an extra config without an url,a canvas button will appear ? but maybe you want something else than this button.

I think we should give up  to try to manage a config for the CanvasSaveButton. Maybe use the `children` property so we will have a definitioon like this in the ShareMenu example:
```
<ShareMenu>
   <CanvasSaveButton />
</ShareMenu> 
```

What do you think ?

Reply:
That makes more sense I agree